sourcegraph/sourcegraph
# Code insights: Add BE insight creation UI

### Context
Since this PR we have some new insight type called insight BE (or over all repositories insight). This insight will collect information from all repositories in sg instance. To introduce this new type of insight this PR adds a new form field to the insight creation/edit UI

<img width="602" alt="image" src="https://user-images.githubusercontent.com/18492575/125697319-e6911903-f070-4cb5-8324-83cd02d026d1.png">

With this turned-on mode, the creation UI creates BE insight. A couple of words about this new BE insights 
- Collect data for insight from all repositories that you have in instance (because of this we disable the repositories field in the form)
- Work with fixed time step between chart data points (2 weeks). Because of that we disable the time step and set 2 weeks-value programmatically 
- This type of insight lives in a special place in the setting file - `insights.allrepos`

```jsonc
"insights.allrepos" : { "<insight_id>: { /* insight configuration */}"}
```

### What have been done
- [x] Add all repos UI (form checkbox)
- [x] Add all repos logic to other form fields (disable repositories and time step in case if all repos is checked
- [x] Connect UI to the update logic

### How to test
1. Go to `/insights/create/search` - the search based insight creation UI page
2. Try to fill some value as you do usually when you want to create a code insight 
3. Turn on all repos mode (checked the new checkbox from screenshot above)
4. You must see that repositories and timestep field are disabled 
5. Click create button
6. Check your settings, you must see `insights.allrepos` map with your newly created insight's configurations
